### ISSUE TITLE
Global dictionary vs. thread safety

### ISSUE THREAD

**Issue by fireice-uk** [breckinloggins/ngtemplate#4]
Hi,
How would you feel about:
a) Getting rid of the global dictionary
b) Converting it to a user-controlled context associated to a template.

Lack of thread safety in this feature prevents the code being used in multi-threaded web servers.

**Comment by breckinloggins** [comment]
Yes, that would be a very logical thing to do. If I'm not mistaken that's
the only shared data so it should be the only thing standing in the way of
thread safety.
